src/main/java/org/jahia/modules/jahiaoauth/impl/JahiaOAuthServiceImpl.java

```diff
@@ -177,6 +175,84 @@ private Map<String, Object> extractAccessTokenData(OAuth2AccessToken accessToken
         return tokenData;
     }
 
+    /**
+     * Extracts all top-level simple properties from a JSONObject into a flat map.
+     * This allows for efficient lookups when checking for dynamically mapped properties not defined in the connector's availableProperties.
+     * Only simple values (strings, numbers, booleans, null) are extracted.
+     * Nested objects and arrays are skipped - they require explicit mapping via connector's availableProperties.
+     * If a property key already exists in the map, it will be overwritten with the new value.
+     *
+     * @param jsonObject the JSON object to extract properties from
+     * @param targetMap  the map to populate with extracted properties
+     */
+    private void extractAllJsonProperties(JSONObject jsonObject, Map<String, Object> targetMap) {
+        Iterator<String> keys = jsonObject.keys();
+        while (keys.hasNext()) {
+            String key = keys.next();
+            try {
+                Object value = jsonObject.get(key);
+
+                // Convert JSONObject.NULL to Java null
+                if (value == JSONObject.NULL) {
+                    value = null;
+                }
+
+                // Only extract simple types: String, Number (Integer, Long, Double), Boolean, or null
+                if (value == null || value instanceof String || value instanceof Number || value instanceof Boolean) {
+                    Object previousValue = targetMap.put(key, value);
+                    if (previousValue != null) {
+                        logger.debug("Property '{}' got overwritten - old value: '{}', new value: '{}'", key, previousValue, value);
+                    }
+                } else {
+                    // Skip complex types (JSONObject, JSONArray, or any other unexpected types)
+                    logger.debug(
+                            "Skipping non-simple property '{}' of type '{}' - use connector's availableProperties with valuePath for complex values",
+                            key, value.getClass().getSimpleName());
+                }
+            } catch (JSONException e) {
+                logger.error("Failed to extract property '{}' from JSON: {}", key, e.getMessage(), e);
+            }
```

Comment:
I think we may consider crashing the execution instead, if we continue current thread execution without props extracted, we could risk inconsistent states.
From my perspective it's better to guarantee integrity and fail instead of silently fail and potentially create inconsistent prop mapped for some users.